### PR TITLE
Auth API documentation

### LINKED ISSUE #13178: Auth API documentation
Things to review:

- URLs (by key name and also /api/v3 prefix, subdomain/username stuff)
- Permissions names (`["update", "select", "trigger", "references", "truncate", "delete", "insert"]`). Might want to fix some on rails side
  - Master key permissions leak existence of analysis caches
- Token lengths
- `type` is not honoured in creation

### PR DESCRIPTION
**Pull request by amartinj**
Closes #13178

### CODE REVIEW

**Comment by amartinj** [comment]
retest this, please

**Comment by amartinj** [comment]
Restest this, please

**Comment by amartinj** [comment]
retest this, please

File: docs/reference/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,518 @@
+openapi: 3.0.0
+info:
+  title: Auth API
+  description: >
+    # Introduction
+
+    This API allows you to manage API keys. API keys are the fundamental
+    building block of Carto's authorization system. See this this guide for more
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: building block of ~Carto's~ CARTO's authorization

File: docs/reference/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+    information.
+
+
+    This API Accepts and returns JSON.
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: This API ~Accepts~ accepts ...

File: docs/reference/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+
+    API base endpoint is `https://<your_username>.carto.com/api/v3/api_keys` or `https://<org_name>.carto.com/u/<your_username/api/v3/api_keys` in case you are using a user belonging to an organization.
+
+    # Authorization
+
+    There are three different API Keys that provide with different access privileges:
+
+    1. `default`: This api provides access to all public objects. It cannot be removed.
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: This ~api~ API key provides...

File: docs/reference/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+                }
+        ```
+        - `database`: Describes to which tables and which privleges on each table this API Key grants access to though `tables` attribute:
+        ```{
+              {
+                    "type": "database",
+                    "tables": [
+                        {
+                            "schema": "public",
+                            "name": "puigdemont",
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: !!!

File: docs/reference/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+                                "update"
+                            ]
+                        }
+                    ]
+                }
+        ```
+
+    # Authentication
+
+    In order to authenticate your requests to the api, they need to include a `Basic` `Authentication` header, where the `username` would be your username and the `password` would be your API Key's token. This authentication method will be valid across all CARTO components (Auth API, Maps API, SQL API). You can build your own `Authorization` header as follows:
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: your requests to the ~api~ API...

File: docs/reference/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+    "Basic #{Base64.strict_encode64(username + ':' + api_key.token)}"
+    ```
+
+
+    **Important:** The API key you provide to access Auth API must be of type
+    `master`.
+  version: 0.0.1
+  contact:
+    name: Have you found an error? Github issues
+    url: 'https://github.com'
```
Comment (alrocar) [inline docs/reference/swagger.yaml]: I'd use this -> https://github.com/CartoDB/cartodb/issues/new

File: docs/reference/swagger.yaml (hunk 4 above)
Comment (amartinj) [reply docs/reference/swagger.yaml]: 🌚

File: docs/reference/swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+    In order to authenticate your requests to the API, they need to include a `Basic` `Authentication` header, where the `username` would be your username and the `password` would be your API Key's token. This authentication method will be valid across all CARTO components (Auth API, Maps API, SQL API). You can build your own `Authorization` header as follows:
+
+
+    ```
+    "Basic #{Base64.strict_encode64(username + ':' + api_key.token)}"
+    ```
+
+
+    **Important:** The API key you provide to access Auth API must be of type
+    `master`.
```
Comment (javitonino) [inline docs/reference/swagger.yaml]: This is no longer true, :laughing:

File: docs/reference/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+                                "update"
+                            ]
+                        }
+                    ]
+                }
+        ```
+
+    # Authentication
+
+    In order to authenticate your requests to the API, they need to include a `Basic` `Authentication` header, where the `username` would be your username and the `password` would be your API Key's token. This authentication method will be valid across all CARTO components (Auth API, Maps API, SQL API). You can build your own `Authorization` header as follows:
```
Comment (javitonino) [inline docs/reference/swagger.yaml]: Do we want to advertise the `?api_key=` parameter as well?

Comment (amartinj) [reply docs/reference/swagger.yaml]: I'd say we don't, do we?

File: docs/reference/swagger.yaml (hunk 7 above)
Comment (amartinj) [reply docs/reference/swagger.yaml]: Good point

**Comment by csubira** [comment]
Included in https://github.com/CartoDB/cartodb/pull/13643

**Pull request closed without merging**